lib/pxlogger.js

```diff
@@ -7,31 +7,33 @@ const { LoggerSeverity } = require('./enums/LoggerSeverity');
 
 class PxLogger {
     constructor(params) {
-        this.loggerEnabled = params.px_logger_severity !== LoggerSeverity.NONE;
-        this.debugMode = params.px_logger_severity === LoggerSeverity.DEBUG || false;
+        this.loggerSeverity = params.px_logger_severity;
         this.appId = params.px_app_id || 'PX_APP_ID';
         this.internalLogger = params.customLogger || console;
+        this.logs = [];
     }
 
     debug(msg) {
-        if (!this.loggerEnabled) {
-            return;
-        }
-        if (this.debugMode && typeof msg === 'string') {
+        this.recordLog(msg, LoggerSeverity.DEBUG);
+        if (this.loggerSeverity === LoggerSeverity.DEBUG && typeof msg === 'string') {
```

Comment:
Doing string comparison on every log is costly - why not keep it as a boolean flag?

Reply:
changed